src/module-elasticsuite-swatches/Block/Navigation/Renderer/Swatches/RenderLayered.php

```diff
@@ -26,6 +27,50 @@
  */
 class RenderLayered extends \Magento\Swatches\Block\LayeredNavigation\RenderLayered
 {
+    /**
+     * @return array
+     */
+    public function getSwatchData(): array
+    {
+        if (false === $this->eavAttribute instanceof Attribute) {
+            throw new \RuntimeException('Magento_Swatches: RenderLayered: Attribute has not been set.');
+        }
+
+        $attributeOptions = [];
+        $attributeOptionsSort = [];
+
+        $sortingArr = [];
+        foreach ($this->filter->getItems() as $item) {
+            $sortingArr[] = $item['label'];
+        }
+
+        foreach ($this->eavAttribute->getOptions() as $option) {
+            if ($currentOption = $this->getFilterOption($this->filter->getItems(), $option)) {
+                $attributeOptions[$option->getLabel()] = array_merge($currentOption, ['id' => $option->getValue()]);
+            } elseif ($this->isShowEmptyResults()) {
+                $attributeOptions[$option->getLabel()] = array_merge($this->getUnusedOption($option), ['id' => $option->getValue()]);
+            }
+        }
+
```

Comment:
3. Hence this last part where you built the array with the attribute options in the expected orders with the attribute option ids this time as keys (because it's a requirement for the `getSwatchesByOptionsId` helper method.